pkg/service/utgen/ai.go

```diff
@@ -107,6 +109,10 @@ func (ai *AIClient) Call(ctx context.Context, prompt *Prompt, maxTokens int) (st
 
 	var apiBaseURL string
 
+	// Create a context that cancels on SIGINT (Ctrl+C) or SIGTERM
+	ctx, cancel := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
+	defer cancel()
+
```

Comment:
the parent context is already handling this pls refer to the code of ctx.go in utils. Thus by passing the parent context cancel we won't have to look at any particular context